pkg/controller/bastion/actuator_reconcile.go

```diff
@@ -0,0 +1,327 @@
+
+	ignitionSecret, err := createIgnitionSecret(namespace, opt)
+	if err != nil {
+		return fmt.Errorf("failed to create ignition secret: %w", err)
+	}
+
+	bastionMachine := createMachine(namespace, a.bastionConfig, infraStatus, opt.BastionInstanceName, ignitionSecret.Name)
+
+	machine, err := applyMachine(ctx, onmetalClient, ignitionSecret, bastionMachine)
+	if err != nil {
+		return fmt.Errorf("failed to create machine: %w", err)
+	}
+
+	endpoints, err := getMachineEndpoints(machine)
+	if err != nil {
+		return fmt.Errorf("failed to get machine endpoints: %w", err)
+	}
+
+	if !endpoints.Ready() {
+		return &reconcilerutils.RequeueAfterError{
+			// requeue rather soon, so that the user (most likely gardenctl eventually)
+			// doesn't have to wait too long for the public endpoint to become available
+			RequeueAfter: 5 * time.Second,
+			Cause:        fmt.Errorf("bastion instance has no public/private endpoints yet"),
+		}
+	}
+
+	// once a public endpoint is available, publish the endpoint on the
+	// Bastion resource to notify upstream about the ready instance
+	log.V(2).Info("Successfully reconciled bastion host")
+	patch := client.MergeFrom(bastion.DeepCopy())
+	bastion.Status.Ingress = endpoints.public
+	return a.Client().Status().Patch(ctx, bastion, patch)
+}
+
+// getMachineEndpoints function returns the bastion endpoints of a running
+// machine. It first validates that the machine is in running state, then
+// extracts the private and public IP of the machine's network interface, and
+// finally converts the IPs to their respective ingress addresses.
+func getMachineEndpoints(machine *computev1alpha1.Machine) (*bastionEndpoints, error) {
+	if machine == nil {
+		return nil, fmt.Errorf("machine can't be nil")
+	}
+
+	if machine.Status.State != computev1alpha1.MachineStateRunning {
+		return nil, fmt.Errorf("machine not running, status: %s", machine.Status.State)
+	}
+
+	endpoints := &bastionEndpoints{}
+
+	if len(machine.Status.NetworkInterfaces) == 0 {
+		return nil, fmt.Errorf("no network interface found for machine: %s", machine.Name)
+	}
+
+	if len(machine.Status.NetworkInterfaces[0].IPs) == 0 {
+		return nil, fmt.Errorf("no private ip found for network interface for machine: %s", machine.Name)
+	}
+
+	privateIP := machine.Status.NetworkInterfaces[0].IPs[0].String()
+	if ingress := addressToIngress(&machine.Name, &privateIP); ingress != nil {
+		endpoints.private = ingress
+	}
+
+	if machine.Status.NetworkInterfaces[0].VirtualIP == nil {
+		return nil, fmt.Errorf("no virtual ip found for network interface for machine: %s", machine.Name)
+	}
+
+	virtualIP := machine.Status.NetworkInterfaces[0].VirtualIP.String()
+	if ingress := addressToIngress(&machine.Name, &virtualIP); ingress != nil {
+		endpoints.public = ingress
+	}
+
+	return endpoints, nil
+}
+
+// bastionConfigCheck checks whether a bastion configuration is valid.
+func bastionConfigCheck(bastionConfig *controllerconfig.BastionConfig) error {
```

Comment:
Just call the variable `config` as we know that we have a bastion host configuration here.